app/javascript/ReadOnlyPreview.jsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+// eslint-disable-next-line react/prefer-stateless-function
+export default class ReadOnlyPreview extends React.Component {
```

Comment:
Same here, any time you've got a class component that's just a `render()` function, it should probably be a functional component.